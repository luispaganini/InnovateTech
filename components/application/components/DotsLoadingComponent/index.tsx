import { Animated, ColorSchemeName } from 'react-native'
import React, { useEffect, useRef } from 'react'
import { Dot, LoadingContainer } from './styles';

export default function DotsLoadingComponent({colorTheme}: {colorTheme: ColorSchemeName}) {
    const opacity = useRef(new Animated.Value(0)).current;

    const animate = () => {
      Animated.sequence([
        Animated.timing(opacity, {
          toValue: 1,
          duration: 500,
          useNativeDriver: true,
        }),
        Animated.timing(opacity, {
          toValue: 0,
          duration: 500,
          useNativeDriver: true,
        }),
      ]).start(() => animate());
    };
  
    useEffect(() => {
      animate();
    }, []);
  
    return (
      <LoadingContainer>
        <Dot colorTheme={colorTheme} style={{ opacity }} testID={'dot-1'}>.</Dot>
        <Dot colorTheme={colorTheme} style={{ opacity }} testID={'dot-2'}>.</Dot>
        <Dot colorTheme={colorTheme} style={{ opacity }} testID={'dot-3'}>.</Dot>
      </LoadingContainer>
    );
}