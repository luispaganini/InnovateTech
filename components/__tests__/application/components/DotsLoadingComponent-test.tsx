import DotsLoadingComponent from '@/components/application/components/DotsLoadingComponent';
import React from 'react';
import renderer, { act } from 'react-test-renderer';

describe('DotsLoadingComponent', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DotsLoadingComponent colorTheme="light" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('animates dots', () => {
    jest.useFakeTimers();

    const component = renderer.create(<DotsLoadingComponent colorTheme="light" />);
    const dot1 = component.root.findByProps({ testID: 'dot-1' });
    const dot2 = component.root.findByProps({ testID: 'dot-2' });
    const dot3 = component.root.findByProps({ testID: 'dot-3' });

    // Verify that the dots initially have opacity 0
    expect(dot1.props.style.opacity._value).toEqual(0);
    expect(dot2.props.style.opacity._value).toEqual(0);
    expect(dot3.props.style.opacity._value).toEqual(0);

    // Advance timers by 500ms (duration of first animation)
    act(() => {
      jest.advanceTimersByTime(500);
      component.update(<DotsLoadingComponent colorTheme="light" />);
    });
  });
});
