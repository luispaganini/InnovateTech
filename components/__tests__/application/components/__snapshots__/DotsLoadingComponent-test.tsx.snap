// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotsLoadingComponent renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    collapsable={false}
    colorTheme="light"
    style={
      {
        "color": "black",
        "fontSize": 50,
        "marginLeft": 5,
        "opacity": 0,
      }
    }
    testID="dot-1"
  >
    .
  </Text>
  <Text
    collapsable={false}
    colorTheme="light"
    style={
      {
        "color": "black",
        "fontSize": 50,
        "marginLeft": 5,
        "opacity": 0,
      }
    }
    testID="dot-2"
  >
    .
  </Text>
  <Text
    collapsable={false}
    colorTheme="light"
    style={
      {
        "color": "black",
        "fontSize": 50,
        "marginLeft": 5,
        "opacity": 0,
      }
    }
    testID="dot-3"
  >
    .
  </Text>
</View>
`;
